plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/githubDeployKey.ts

```diff
@@ -0,0 +1,159 @@
+import { getOctokitOptions } from './helpers';
+import { Octokit } from 'octokit';
+import Sodium from 'libsodium-wrappers';
+
+/**
+ * Creates an `github:api:execute` Scaffolder action.
+ *
+ * @remarks
+ *
+ * @public
+ */
+export function createGithubDeployKeyAction(options: {
+  integrations: ScmIntegrationRegistry;
+}) {
+  const { integrations } = options;
+  // For more information on how to define custom actions, see
+  //   https://backstage.io/docs/features/software-templates/writing-custom-actions
+  return createTemplateAction<{
+    repoUrl: string;
+    publicKey: string;
+    privateKey: string;
+    deployKeyName: string;
+    privateKeySecretName?: string;
+    token?: string;
+  }>({
+    id: 'github:deployKey:create',
+    description: 'Creates and stores Deploy Keys',
+    schema: {
+      input: {
+        type: 'object',
+        required: ['repoUrl', 'publicKey', 'privateKey', 'deployKeyName'],
+        properties: {
+          repoUrl: {
+            title: 'Repository Location',
+            description: `Accepts the format 'github.com?repo=reponame&owner=owner' where 'reponame' is the new repository name and 'owner' is an organization or username`,
+            type: 'string',
+          },
+          publicKey: {
+            title: 'SSH Public Key',
+            description: `Generated from ssh-keygen.  Begins with 'ssh-rsa', 'ecdsa-sha2-nistp256', 'ecdsa-sha2-nistp384', 'ecdsa-sha2-nistp521', 'ssh-ed25519', 'sk-ecdsa-sha2-nistp256@openssh.com', or 'sk-ssh-ed25519@openssh.com'.`,
+            type: 'string',
+          },
+          privateKey: {
+            title: 'SSH Private Key',
+            description: `SSH Private Key generated from ssh-keygen`,
+            type: 'string',
+          },
+          deployKeyName: {
+            title: 'Deploy Key Name',
+            description: `Name of the Deploy Key`,
+            type: 'string',
+          },
+          privateKeySecretName: {
+            title: 'Private Key GitHub Secret Name',
+            description: `Name of the GitHub Secret to store the private key related to the Deploy Key.  Defaults to: 'KEY_NAME_PRIVATE_KEY' where 'KEY_NAME' is the name of the Deploy Key`,
+            type: 'string',
+          },
+          token: {
+            title: 'Authentication Token',
+            type: 'string',
+            description: 'The token to use for authorization to GitHub',
+          },
+        },
+      },
+      output: {
+        type: 'object',
+        properties: {
+          privateKeySecretName: {
+            title: 'The GitHub Action Repo Secret Name for the Private Key',
+            type: 'string',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const {
+        repoUrl,
+        publicKey,
+        privateKey,
+        deployKeyName,
+        privateKeySecretName = `${deployKeyName
+          .split(' ')
+          .join('_')
+          .toLocaleUpperCase()}_PRIVATE_KEY`,
```

Comment:
```suggestion
          .toLocaleUpperCase('en-US')}_PRIVATE_KEY`,
```

you will get odd effects in some locales otherwise ...